packages/protocol/contracts/L2/Lib1559Math.sol

```diff
@@ -36,41 +36,57 @@ library Lib1559Math {
         // block, however, this block's gas used will affect the next
         // block's base fee.
         basefee_ = basefee(gasExcess_, uint256(_adjustmentQuotient) * _gasTarget);
-
-        // Always make sure basefee is nonzero, this is required by the node.
-        if (basefee_ == 0) basefee_ = 1;
     }
 
-    /// @dev eth_qty(excess_gas_issued) / (TARGET * ADJUSTMENT_QUOTIENT)
-    /// @param _gasExcess The gas excess value
-    /// @param _adjustmentFactor The product of gasTarget and adjustmentQuotient
-    function basefee(
-        uint256 _gasExcess,
-        uint256 _adjustmentFactor
+    /// @dev Returns the new gas excess that will keep the basefee the same.
+    /// `_newGasTarget * ln(_newGasTarget / _target) + _gasExcess * _newGasTarget / _target`
+    function adjustExcess(
+        uint64 _gasExcess,
+        uint64 _gasTarget,
+        uint64 _newGasTarget
```

Comment:
Maybe slightly easier to take as input the current basefee (which we can already calculate)? Then the formula should just be `excess_gas = _newGasTarget * ln(basefee * _newGasTarget)`.

Is the reason for the unchecked code to avoid over/under flow? Because normally this code should be trigger very rarely I think so saving some gas shouldn't be important.

Reply:
The `uncheck` can be removed as the function shall be rarely used.

Reply:
> Maybe slightly easier to take as input the current basefee (which we can already calculate)? Then the formula should just be `excess_gas = _newGasTarget * ln(basefee * _newGasTarget)`.

This will work. However, to calculate the parent block's basefee, we need both the parent's `gasIssuancePerSecond` and `basefeeAdjustmentQuotient` which are not available in the current block. While using the current implementation we only need their product (`parentGasTarget`).